File: level1JS/GcdLcm.js
function solution(n, m) {
    let divider = 1;

    for(let i = 1; i <= Math.min(n, m); i++) {
        if(n % i === 0 && m % i === 0) divider = i;
    }
    const multiple = (n / divider) * (m / divider) * divider;
    const answer = [divider, multiple];
    return answer;
}

console.log(solution(1, 1));

File: level1JAVA/src/GcdLcm.java
import java.util.Arrays;

class GcdLcm {
    public int[] solution(int n, int m) {
        int max = Math.max(n, m);
        int min = Math.min(n, m);
        int gcd = getGCD(max, min);
        int lcm = (max * min) / gcd;
        return new int[]{gcd, lcm};
    }
    // 유클리드 호제법
    public int getGCD(int max, int min) {
        int rest = max % min;
        if(rest == 0) {
            return min;
        }
        return getGCD(min, rest);
    }

    public static void main(String[] args) {
        GcdLcm gcdLcm = new GcdLcm();
        int[] result = gcdLcm.solution(3, 12);
        System.out.println("result = " + Arrays.toString(result));
    }
}
